src/main/java/org/micks/champmaker/meals/MealService.java

```diff
@@ -0,0 +1,40 @@
+package org.micks.champmaker.meals;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.persistence.EntityNotFoundException;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class MealService {
+
+    @Autowired
+    MealRepository mealRepository;
+
+    public void createMeal(long champId, MealDTO mealDTO) {
+        MealEntity mealEntity = new MealEntity(champId, mealDTO.getName(), mealDTO.isVegeOption());
+        mealRepository.save(mealEntity);
+    }
+
+    public void editMeals(long champId, MealDTO mealDTO) throws EntityNotFoundException {
+        Optional<MealEntity> optionalMeal = mealRepository.findById(champId);
```

Comment:
Wrong id - you should search by `mealId`, it needs to be added on controller level (see my other comment) and passed here to service level too.